fastjet/internal/delaunay/delaunay.go

```diff
@@ -162,6 +169,36 @@ func (d *Delaunay) insertInside(p *Point, t *Triangle) []*Point {
 // insertOnEdge inserts a point on an Edge between two triangles. It returns the points
 // whose nearest neighbor changed.
 func (d *Delaunay) insertOnEdge(p *Point, t *Triangle) []*Point {
+	// Check if p is a duplicate
+	switch {
+	case p.Equals(t.A):
+		if p.id == t.A.id {
+			panic(fmt.Errorf("Point %v was previously inserted", p))
+		}
+		p.nearest = t.A
+		p.dist2 = 0
+		t.A.nearest = p
+		t.A.dist2 = 0
+		return []*Point{p, t.A}
+	case p.Equals(t.B):
+		if p.id == t.B.id {
+			panic(fmt.Errorf("Point %v was previously inserted", p))
+		}
+		p.nearest = t.B
+		p.dist2 = 0
+		t.B.nearest = p
+		t.B.dist2 = 0
+		return []*Point{p, t.B}
+	case p.Equals(t.C):
+		if p.id == t.C.id {
+			panic(fmt.Errorf("Point %v was previously inserted", p))
```

Comment:
please prepend a `"delaunay: ` string to the panic message.

Reply:
Done